BranchSDK/src/BranchIO/Branch.cpp

```diff
@@ -179,6 +180,30 @@ Branch::sendEvent(const Event &event, IRequestCallback *callback) {
     getRequestManager()->enqueue(event, callback);
 }
 
+void
+Branch::setIdentity(const std::string& userId, IRequestCallback *callback) {
+    if (getSessionInfo().hasSessionId()) {
+        IdentityLoginEvent event(userId);
+        sendEvent(event, callback);
+    } else {
+        if (callback != NULL) {
+            callback->onError(0, 0, "No Session");
```

Comment:
Error message could be more elaborate, like "No session, must send Install or Open event first."

Reply:
Updated.  I generally don't want to put source code implementation details into callback "reason" strings.   I updated this to `No Session has been started.`